include/mbedtls/ssl.h

```diff
@@ -607,22 +621,20 @@ typedef int mbedtls_ssl_async_sign_t( void *connection_ctx,
  *                  this function sends or enqueues a request and does
  *                  not wait for the operation to complete.
  *
- *                  The parameters \c connection_ctx and \c cert are
+ *                  The parameters \c ssl and \c cert are
  *                  guaranteed to remain valid as long as the SSL
  *                  configuration remains valid. On the other hand, this
  *                  function must save the contents of \c hash, as the
  *                  \c hash buffer is no longer valid when this function
  *                  returns.
  *
- * \param connection_ctx  Pointer to the connection context set in the
- *                        SSL configuration
- * \param p_operation_ctx On success, pointer to the operation context.
- *                        This must be a non-null pointer. Success means
- *                        that an operation was started, and the return
- *                        status is 0 or \c MBEDTLS_ERR_SSL_ASYNC_IN_PROGRESS.
- *                        This pointer will be passed to later calls to the
- *                        resume or cancel function. If the callback fails,
- *                        the value is ignored.
+ *                  This function may call mbedtls_ssl_async_set_data() to
```

Comment:
Since there is both `config_data` and `data`, I'm afraid there is a risk of users being slightly confused. Perhaps renaming this one to `operation_data` could help, or we can leave it as this and just rely on documentation to avoid any possible confusion.

Reply:
Not sure what you're suggesting here. What's called `data`?

Reply:
I mean there's `mbedtls_ssl_async_get_data()` and `mbedtls_conf_get_async_config_data()`. Since "data" is a very vague term (which is justified here as it's an opaque thing), I think it's usually better to qualify it, especially when there are multiple things we call "data". Technically having only n-1 of them explicitly qualified it's enough (here the absence of qualification means it's not `config_data` so it's the other one as they are only two of them), but I think explicit is better than implicit and `mbedtls_ssl_async_get_operation_data()` might be a better name.

OTOH, now that I've finished reading everything, I became clear to me that the documentation is very consistently stating "operation data" every time `mbedtls_ssl_async_get/set_data()` is mentioned, so the naming is not a blocker for me as the documentation compensates for it.

Reply:
I renamed `mbedtls_ssl_async_get_data` to `mbedtls_ssl_get_async_operation_data` (and likewise for `set`) so the function is now obviously about “async operation data”.